### PR TITLE
replace ecbuild with CMake in CRTM

### LINKED ISSUE #52: switching from ecbuild to cmake
Title: Migrate build system from ecbuild to CMake

---

**Is your feature request related to a problem? Please describe.**

As the developer team, we're finding it increasingly difficult to maintain our build system using ecbuild, due to its less familiar syntax and fewer resources compared to CMake. This hinders our productivity and slows down the progress of our project.

**Describe the solution you'd like**

We would like to transition from ecbuild to CMake. CMake has a more standard syntax and more community resources available, making it easier to use and troubleshoot. It would simplify the build process for developers, leading to a more efficient and effective development process.

**Describe alternatives you've considered**

As an alternative, we've considered continuing with ecbuild, while investing in more in-depth training and resources for our team. However, considering the overall benefits of CMake, migrating seems like the more beneficial choice in the long run.

**Additional context**

This will be a substantial project, so it's recommended to divide it into manageable stages:

1. Research and planning - Understand the current ecbuild setup and plan out how we will recreate this in CMake.
2. Initial conversion - Start creating the CMake setup based on the plan.
3. Testing and debugging - Thoroughly test the new CMake build system and fix any bugs that arise.
4. Documentation - Update all relevant documentation to reflect the change from ecbuild to CMake.
5. Deployment - Roll out the CMake build system to all developers and provide any necessary training.
6. Post-implementation review - After a few weeks or months, gather feedback to understand how well the new system is working and make any necessary adjustments.

It would be helpful if team members could comment with any specific challenges they've had with ecbuild that we should be sure to address in the CMake implementation.

### PR DESCRIPTION
**Pull request by BenjaminTJohnson**
## Description

See https://github.com/JCSDA/CRTMv3/pull/90

This branch is essentially the same, but somehow I messed up the other branch with merging.  

Short story:  Replaces all ecbuild items with CMake.   Removes legacy build system (autotools).   Appears to work correctly with jedi/ufo.

## Issue(s) addressed

Resolves #52 

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,34 +1,115 @@
...
 
-## Ecbuild integration
-find_package( ecbuild QUIET )
-include( ecbuild_system NO_POLICY_SCOPE )
-ecbuild_declare_project()
-list( APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake )
-set( CMAKE_DIRECTORY_LABELS ${PROJECT_NAME} )
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+set(CMAKE_DIRECTORY_LABELS ${PROJECT_NAME})
+#if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
```
Comment (climbfuji) [inline CMakeLists.txt]: Do you need the commented lines 9 and 11?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,34 +1,115 @@
...
+  "${CMAKE_CURRENT_BINARY_DIR}/crtm-config-version.cmake"
+  VERSION ${PROJECT_VERSION}
+  COMPATIBILITY AnyNewerVersion
+)
+
+# Install the generated or configured files
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/crtm-config.cmake"
+              "${CMAKE_CURRENT_BINARY_DIR}/crtm-config-version.cmake"
+        DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+#add_dependencies(${PROJECT_NAME} install)
```
Comment (climbfuji) [inline CMakeLists.txt]: remove this comment?

**Review by climbfuji — APPROVED** [review]
Looks familiar to me, thanks for making these changes. Hopefully this can serve as a boilerplate for other codes that do not use ecbuild but that we need to pull into jedi-bundle.

**Review by chengdang — APPROVED** [review]
All tests passed with cmake!
```
100% tests passed, 0 tests failed out of 160

Label Time Summary:
CRTM_Tests       = 225.32 sec*proc (160 tests)
crtm_tests       = 225.32 sec*proc (160 tests)
download_data    =  62.21 sec*proc (1 test)
executable       =  47.96 sec*proc (2 tests)
script           = 115.16 sec*proc (157 tests)

Total Test time (real) = 229.91 sec
```

**Pull request merged**